src/import-xml/import-xml.service.ts

```diff
@@ -106,6 +110,50 @@ export class ImportXmlService {
     };
   }
 
+  async reprocessXML(professorName: string) {
```

Comment:
Use a name to identify a person is not a good pratice. There are many cases of researchers with the same name. Please change every place (including frontend) to use it as the real id.